app/models/ability.rb

```diff
@@ -0,0 +1,17 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new
```

Comment:
ideally you should check if user is defined
and if yes
then use user_ability
else guest ability
no need to build new user

user ability = user ablility + guest ability